### DISCUSSION TITLE
msys2

### DISCUSSION THREAD

**Discussion by DaveAllerton** [vsg-dev/VulkanSceneGraph#1015] [General]
Has anyone managed to install VulkanSceneGraph on msys2 (mingw64)?

**Comment by robertosfield** [comment]
I don't recall discussion about msys2/mingw64 but perhaps someone has tried compiling it, hopefully Windows/MinGW users here will pop up and add there experiences/suggestions.

Looking at the VSG codebase I only see references to __MINGW32__ so perhaps 64 bit won't work yet.  These are the gits for mingw I get with a search via our github repo:

https://github.com/search?q=repo%3Avsg-dev%2FVulkanSceneGraph%20mingw&type=code

Given there isn't many instances even if code changes need to be made for 64bit builds it shouldn't be difficult to add - most platforms build 64bit so all the C++ side should work straight away, it's just the IO side that might need some minor tweaks.

I work just under Linux so can't provide any direct support for testing/porting on other platforms, so I rely upon members of the community to help build and test on the their preferred platforms.

**Reply by maisvendoo** [reply]
Build in msys2 enveronment was failure, in my case. But I built VSG 1.0.9 and VSG 1.1.0 by mingw 11.2 64-bit. This task was difficult. I used mingw toolchain from Qt 6.3.1 distribution and VulkanSDK extracted from msys2 packages. And it's works

![12a0f1208a5278903eec12a9f92bda9f](https://github.com/vsg-dev/VulkanSceneGraph/assets/5387928/d8d8e2cc-4129-4f2f-9543-dcb2e482e0a5)

**Comment by timoore** [comment]
I can't comment on mingw64, but I have used two alternatives:

* MSVC plus the Git for Windows tools. This is my daily driver at the moment, sigh. I don't understand the precise relationship between Git for Windows and msys2, but I have used GfW as my bash shell on Windows for several years and am generally happy with it. You can run builds from a command line, though I usually don't.

* WSL2 plus Microsoft's Vulkan-on-D3D mesa contribution (dozen). I screwed around with this enough to get the Vulkan SDK cube running. I haven't actually tried VSG yet, but the Vulkan features seem complete enough to be able to run VSG. Performance is said to be reasonable. I don't know why you would use this instead of native Windows or Linux, unless you need to test on Linux and can't  install it as a second OS. That's sort of my use case: I have an Asus ROG Ally handhold on which I'm very reluctant to install Linux, but I do want to be able to test things on Ubuntu.
